extern/sector-storage/worker_local.go

```diff
@@ -482,41 +488,164 @@ func (l *LocalWorker) Paths(ctx context.Context) ([]stores.StoragePath, error) {
 	return l.localStore.Local(ctx)
 }
 
-func (l *LocalWorker) Info(context.Context) (storiface.WorkerInfo, error) {
-	hostname, err := os.Hostname() // TODO: allow overriding from config
+func cgroupV2Path() (string, error) {
+	buf, err := ioutil.ReadFile("/proc/self/cgroup")
 	if err != nil {
-		panic(err)
+		return "", err
 	}
+	cgroupPathElems := bytes.Split(buf, []byte{':'})
+	cgroupRelPath := string(bytes.TrimSpace(cgroupPathElems[len(cgroupPathElems)-1]))
 
-	gpus, err := ffi.GetGPUDevices()
+	cgroupMountPoint, err := cgroupV2MountPoint()
+	if err != nil || cgroupMountPoint == "" {
+		return "", err
+	}
+	return cgroupMountPoint + cgroupRelPath, nil
+}
+
+func cgroupV2MountPoint() (string, error) {
+	f, err := os.Open("/proc/self/mountinfo")
 	if err != nil {
-		log.Errorf("getting gpu devices failed: %+v", err)
+		return "", err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		fields := bytes.Fields(scanner.Bytes())
+		if len(fields) >= 9 && bytes.Equal(fields[8], []byte("cgroup2")) {
+			return string(fields[4]), nil
+		}
+	}
+	return "", nil
+}
```

Comment:
We already use `github.com/containerd/cgroups` for getting memory info in go-watchdog, so it would be nice to use this here as well.

See https://github.com/raulk/go-watchdog/blob/master/watchdog_linux.go#L29-L46

Should also be disabled on non-linux builds (though I see we're ignoring the error below and falling back to host info, so that's maybe fine.)

Reply:
I've changed around this logic in the latest commit. My method only supported cgroups v2, now it supports v1 and v2. Unfortunately containerd/cgroups v2 support is missing a few features, like getting the mount point so I had to still implement that myself.